File: frontend/src/app/models/tasks.ts
import {Job, Priority, Status} from "./job";
import {User} from "./user";
import {Contact} from "./contact";
import {Company} from "./company";
import {Campaign} from "./campaign";
import {TaskNotes} from "./note";
import {LeadTasks} from "./lead";

export interface Tasks {
  id: number;
  taskTitle: string;
  taskDescription: string;
  dueDate: Date;
  status: Status;
  priority: Priority;
  assignedToUserId: string;
  assignedToUser: User;
  campaignId: number;
  campaign: Campaign;
  jobId: number;
  job: Job;
  contactId: number;
  contact: Contact;
  companyId: number;
  company: Company;
  dateCreated: Date;
  dateUpdated?: Date;
  dateCompleted?: Date;
  taskNotes?: TaskNotes[];
  jobTasks?: JobTasks[];
  leadTasks?: LeadTasks[];
  campaignTasks?: CampaignTasks[];
  companyTasks?: CompanyTasks[];
}

export interface JobTasks {
  id: number;
  taskId: number;
  jobId: number;
  job: Job;
  tasks: Tasks;
}

export interface CampaignTasks {
  id: number;
  taskId: number;
  campaignId: number;
  campaign: Campaign;
  tasks: Tasks;
}

export interface UserTasks {
  id: number;
  taskId: number;
  userId: string;
  tasks: Tasks;
  user: User;
}

export interface ContactTasks {
  id: number;
  taskId: number;
  contactId: number;
  contact: Contact;
  tasks: Tasks;
}
export interface CompanyTasks {
  id: number;
  companyId: number;
  taskId: number;
  tasks: Tasks;
  company: Company;
}

File: backend/backend/Areas/Main/Models/Tasks.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using backend.Areas.Identity.Models;
using backend.Areas.Main.Models.Enums;

namespace backend.Areas.Main.Models;

public class Tasks
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }
    public string TaskTitle { get; set; }
    public string TaskDescription { get; set; }
    public DateTime DueDate { get; set; }
    public Status Status { get; set; } 
    public Priority Priority { get; set; } 
    public string AssignedToUserId { get; set; } 
    [ForeignKey(nameof(AssignedToUserId))]
    public virtual User AssignedToUser { get; set; } 
    public DateTime DateCreated { get; set; }
    public DateTime? DateUpdated { get; set; }
    public DateTime? DateCompleted { get; set; } 
    public IEnumerable<TaskNotes>? TaskNotes { get; set; }
    public IEnumerable<JobTask>? JobTasks { get; set; }
    public IEnumerable<LeadTask>? LeadTasks { get; set; }
    public IEnumerable<CampaignTasks>? CampaignTasks { get; set; }
    public IEnumerable<CompanyTask>? CompanyTasks { get; set; }
    public IEnumerable<UserTasks>? UserTasks { get; set; }
    public IEnumerable<ContactTasks>? ContactTasks { get; set; }
    
}